File: src/frontend/src/api/orders/create-order-line-item-api-request.ts
import { CreateRouteApiRequest } from "./create-route-api-request";

export interface CreateOrderLineItemApiRequest {
  route: CreateRouteApiRequest;
  materialName: string;
  materialUnit: string;
  materialQuantity: number;
  materialWeight: number;
  materialDimensions: string;
}

File: src/services/ordering/Ordering/API/Controllers/V1/Models/CreateOrderLineItemApiRequest.cs
namespace LoadLogic.Services.Ordering.API.Controllers.V1.Models
{
    public class CreateOrderLineItemApiRequest
    {
        public CreateRouteApiRequest? Route { get; set; }
        public string MaterialName { get; set; } = string.Empty;
        public string MaterialUnit { get; set; } = string.Empty;
        public double MaterialQuantity { get; set; }
        public double MaterialWeight { get; set; }
        public string MaterialDimensions { get; set; } = string.Empty;
    }
}
